flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -226,9 +230,21 @@ public static CuratorFramework startCuratorFramework(Configuration configuration
         }
 
         CuratorFramework cf = curatorFrameworkBuilder.build();
-
+        // This handler is only used to handle the error during start phase, and should be
+        // removed after start curator success.
+        UnhandledErrorListener unhandledErrorListener =
+                (message, throwable) -> {
+                    LOG.error(
+                            "Exiting process for unhandled error in start curator framework, "
+                                    + "error message: {}, exiting code: {}",
+                            message,
+                            ZOOKEEPER_FAILURE_EXIT_CODE,
+                            throwable);
+                    System.exit(ZOOKEEPER_FAILURE_EXIT_CODE);
+                };
+        cf.getUnhandledErrorListenable().addListener(unhandledErrorListener);
```

Comment:
I am actually wondering whether we shouldn't only register this one handler and remove the listeners in the `ZooKeeperLeaderElectionDriver` and `ZooKeeperLeaderRetrievalDriver` because these two listeners would be called for every unhandled exception that occurs in Curator independent of the actual source. Moreover, the current `UnhandledErrorListener` always call the `FatalExitExceptionHandler` eventually.

Reply:
I don't really get your meaning， may be it's `should only` ? 

I think we could safely remove the listener in `leaderElectionDriver` and `leaderRetrievalDriver` like you said that，all the three handlers always exit the process now, after one execute other will not be executed. So there is no meaning to register other handlers.

Reply:
Yes, I meant to only register this one failure handler and remove the listener in `leaderElectionDriver` and `leaderRetrievalDriver`.